desktop/app/menu-bar.js

```diff
@@ -3,21 +3,29 @@ import {resolveImage, resolveRootAsURL} from '../resolve-root'
 import hotPath from '../hot-path'
 import menubar from 'menubar'
 
-let color = 'white'
-let platform = ''
-
-if (process.platform === 'darwin') {
-  color = (systemPreferences && systemPreferences.isDarkMode()) ? 'white' : 'black'
-} else if (process.platform === 'win32') {
-  color = 'black'
-  platform = 'windows-'
-}
+let iconType: 'regular' | 'update' | 'badged' = 'regular'
+
+const getIcon = () => { // eslint-disable-line arrow-parens
```

Comment:
we need to store the state of the icon somewhere. it currently gets mutated by the ipc calls to badge / etc. we also sometimes need to update, yet keep the state when osx changes from light to dark mode. so we can't pass this around, or if we did we'd just have to stash it somewhere , just to pass it down